docs/ECMAScriptModules.md

```diff
@@ -38,4 +38,35 @@ import.meta.jest.useFakeTimers();
 // jest === import.meta.jest => true
 ```
 
+Additionally, since ESM evaluates static `import` statements before looking at the code, hoisting on `jest.mock` calls that happens in CJS modules won't work in ESM. To `mock` modules in ESM, you need to use dynamic `import()` after `jest.mock` calls to load the mocked modules, same applies to modules which have to load the mocked modules.
+
+```js
+// - main.cjs
+const { BrowserWindow, app } = require('electron');
+
+// etc.
+
+module.exports = { example };
+```
+
+```js
+// - main.test.js
+import { jest } from '@jest/globals';
+
+jest.mock('electron', () => ({
+  app: {
+    on: jest.fn(),
+    whenReady: jest.fn(() => Promise.resolve()),
+  },
+  BrowserWindow: jest.fn().mockImplementation(() => ({
+    // partial mocks.
+  }))
+}));
+
+const { BrowserWindow } = (await import('electron')).default;
```

Comment:
Hm.. Felt like `require` look better for this case. That’s just style, I’m fine with anything what is working (;

Reply:
I'm fine to go with either style there, just that if we go `require` for `electron` then we should do the same with `main.cjs` as Electron still doesn't work with ESM yet. Tbh I'm feeling a bit strange to use something that only works with CJS as an example here but just that I was working on a Electron project in TypeScript so I used it in my minimal repo for my issue 🤔